src/script/printer/doc_printer.cc

```diff
@@ -572,6 +558,51 @@ void PythonDocPrinter::PrintStmtBlock(const Array<StmtDoc>& docs) {
   DecreaseIndent();
 }
 
+void PythonDocPrinter::PrintBinaryOp(OperationDocNode::Kind operation_kind, const ExprDoc& a,
+                                     const ExprDoc& b) {
+  using OpKind = OperationDocNode::Kind;
+
+  PrintDoc(a);
+  switch (operation_kind) {
+    case OpKind::kAdd:
+      output_ << " + ";
+      break;
+    case OpKind::kSub:
+      output_ << " - ";
+      break;
+    case OpKind::kMul:
+      output_ << " * ";
+      break;
+    case OpKind::kFloorDiv:
+      output_ << " // ";
+      break;
+    case OpKind::kFloorMod:
+      output_ << " % ";
+      break;
+    default:
+      LOG(FATAL) << "Unknown Binary Operation " << static_cast<int>(operation_kind);
+  }
+  PrintDoc(b);
+}
+
+void PythonDocPrinter::PrintSpecialOp(OperationDocNode::Kind operation_kind,
+                                      const Array<ExprDoc>& operands) {
+  using OpKind = OperationDocNode::Kind;
+
+  switch (operation_kind) {
+    case OpKind::kAssert:
```

Comment:
Should we just make a separate `AssertDoc` which is a `StmtDoc`? It seems strange to mix it with operations which are expressions, not statements.

Reply:
That's a good point and I am open to discussion here. When we designed Doc, our goal is to have a minimal representation of language syntax. So the Doc sub classes are either:
1. General syntax structure shared by many ordinary programming languages, like AttrAccessDoc, IfDoc, etc.
2. Critical structure for our use case (representation of TIR), like ScopeDoc and LambdaDoc.

I don't feel that assert belongs to either cases above so it's not represented as a top level Doc class. I agree that it's not intuitive to represent assert statement as an `ExprStmtDoc(OperationDoc(OpKind::kAssert, ...))`.